### ISSUE TITLE
Expected property shorthand warning on computed property

### ISSUE THREAD

**Issue by kikoanis** [eslint/eslint#6015]
**What version of ESLint are you using?**
2.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint = 6.0.4

**Please show your full configuration:**
```
rules: {
   ...
   "object-shorthand": 1,
   ...
}
```
**What did you do? Please include the actual source code causing the issue.**
```
const constant = 'foo'
....
const obj = { [constant]: constant }
```

**What did you expect to happen?**
This should pass or I am missing something on how to do it.
**What actually happened? Please include the actual, raw output from ESLint.**
warning  Expected property shorthand  object-shorthand

**Comment by kaicataldo** [comment]
Can you share the config you're using? Do you have the object-shorthand rule turned on? With the information you provided, it looks like you're not using the object shorthand syntax and, assuming you have the rule turned on in your config, this warning is the expected behavior. The shorthand notation would be:
```js
const constant = 'foo'
....
const obj = { 
  constant
}
```

**Comment by michaelficarra** [comment]
@kaicataldo Look again at the example. The property name is computed.

**Comment by kaicataldo** [comment]
Ah, I see. Shouldn't be doing this so late.

**Comment by kikoanis** [comment]
Updated my original comment to add the configuration part that is responsible for object-shorthand. I do not see any issue here. I only encountered this after upgrading to 2.9.0 and I suspect it is related to this [e2ad1ec](https://github.com/eslint/eslint/commit/e2ad1ec7f6e48f8f4ab9014e5562ec168d9d2442)

**Comment by kaicataldo** [comment]
@kikoanis Yeah, sorry, I shouldn't have been trying to triage so late in the evening. It does seem like a bug introduced in this commit: https://github.com/eslint/eslint/commit/e2ad1ec.

**Comment by kaicataldo** [comment]
PR opened to fix this.

**Comment by 252405649** [comment]
good
